gobblin-metrics/src/main/java/gobblin/metrics/influxdb/InfluxDBReporter.java

```diff
@@ -121,50 +125,133 @@ public static Builder builder(String name) {
     return new Builder(name);
   }
 
-  private Serie.Builder addGauge(String name, Gauge gauge, long timeStamp, Serie.Builder builder) {
-    return builder.values(timeStamp, name, gauge.getValue());
+  private void reportGauge(MetricContext context, String name, Gauge gauge, long timeStamp) {
+    this.influxDB.write(
```

Comment:
Each write results in a HTTP request. Writing individually means more HTTP requests. However, writing individually also gives it a chance to write to different shards, which will be good for write performance.